src/enclave/rpcsessions.h

```diff
@@ -43,20 +43,17 @@ namespace enclave
       rpc_map(rpc_map_)
     {}
 
-    void add_cert(
-      const std::string& sni, CBuffer ca_cert, CBuffer cert, const tls::Pem& pk)
+    void set_cert(CBuffer cert_, const tls::Pem& pk)
     {
       std::lock_guard<SpinLock> guard(lock);
-      auto hasCa = ca_cert != nullb;
-      auto the_cert = std::make_shared<tls::Cert>(
-        sni,
-        hasCa ? std::make_shared<tls::CA>(ca_cert) : nullptr,
-        cert,
+
+      cert = std::make_shared<tls::Cert>(
+        nullptr, // Caller authentication is done by each frontend by looking up
+                 // the caller's certificate in the relevant store table
+        cert_,
         pk,
         nullb,
-        hasCa ? tls::auth_required : tls::auth_optional);
-
-      certs.push_back(std::move(the_cert));
+        tls::auth_optional);
```

Comment:
Can we add a comment on this, to convince me that either we don't always to need to verify or we always check the verification later?

Reply:
I would have thought we needed tls::auth_required here? It's true we check the cert later, but do we even want to allow unauthed connections at all?

Reply:
`tls::auth_required` maps to `MBEDTLS_SSL_VERIFY_REQUIRED`. As per https://tls.mbed.org/api/ssl_8h.html#a5695285c9dbfefec295012b566290f37, the server does not have to verify the client certificate (default). Otherwise, we would have to have the clients certs signed by a trusted CA and replace `nullptr` by that CA here.